src/mca/gds/gds.h

```diff
@@ -289,10 +289,13 @@ typedef pmix_status_t (*pmix_gds_base_module_add_nspace_fn_t)(const char *nspace
                                                               size_t ninfo);
 
 /* define a convenience macro for add_nspace based on peer */
-#define PMIX_GDS_ADD_NSPACE(s, p, n, i, ni)                 \
+#define PMIX_GDS_ADD_NSPACE(s, n, i, ni)                    \
     do {                                                    \
-        pmix_gds_base_module_t *_g = (p)->nptr->compat.gds; \
-        (s) = _g->add_nspace(n, i, ni);                     \
+        pmix_gds_base_active_module_t *_g;                  \
+        PMIX_LIST_FOREACH(_g, &pmix_gds_globals.actives,    \
+                          pmix_gds_base_active_module_t) {  \
+            (s) = _g->module->add_nspace(n, i, ni);         \
```

Comment:
Sorry about that - the comment didn't go where I put it. Anyway, I was just suggesting that you check for NULL  before calling the function